app/controllers/admin/adoption_applications_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::AdoptionApplicationsController < ApplicationController
+
+    def show
+        @adoption_app = AdoptionApplication.find(params[:id])
+        @app_pets = @adoption_app.pets
+        @pets = Pet.search(params[:search])
+
+        if params[:approved_pet].present?
+            pet_id = params[:approved_pet]
```

Comment:
A possible refactor: Consider If you changed the structure of your params so that you passed "approve" or "reject" as the value for a key, and the key could be called params[:update_status] or something generic. Then, you could just check for the presence of that parameter, and sending the value of that parameter (either "approve" or "reject") to a more generic model method that could have the if statement to check whether it should change to approved or rejected. This is an option for pushing some of that logic into a model method instead of a controller. It's totally okay to have if statements in the controller, but this is a case where you could simplify the controller  a bit and just have one if statement instead of 2